power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -48,7 +48,7 @@ template <bool is_const> class DataPointer {
 
     template <class T> std::pair<ptr_t<T>, ptr_t<T>> get_iterators(Idx pos) const {
         assert(pos < batch_size_);
-        ptr_t<T> const ptr = reinterpret_cast<ptr_t<T>>(ptr_);
+        auto* const ptr = reinterpret_cast<ptr_t<T>>(ptr_);
```

Comment:
did the code smell explicitly say that you should use `auto*` here? (if so, that's great but i would expect `auto` to be sufficient)